agent/src/main/java/com/microsoft/hydralab/agent/runner/TestRunner.java

```diff
@@ -41,12 +48,29 @@ public void runTestOnDevice(TestTask testTask, DeviceInfo deviceInfo, Logger log
         setUp(deviceInfo, testTask, deviceTestTask);
         checkTestTaskCancel(testTask);
 
+        PerformanceInspectionService performanceInspectionService = new PerformanceInspectionService(deviceTestTask.getDeviceTestResultFolder());
+        if (testTask.isEnableBatteryTest()) {
+            performanceInspectionService.addInspector(performanceManager.getAndroidBatteryInspector());
+            performanceInspectionService.addInspector(performanceManager.getWindowsBatteryInspector());
+        }
+        if (testTask.isEnableMemoryTest()) {
+            performanceInspectionService.addInspector(performanceManager.getAndroidMemoryDumpInspector());
+            performanceInspectionService.addInspector(performanceManager.getAndroidMemoryInfoInspector());
+            performanceInspectionService.addInspector(performanceManager.getWindowsMemoryInspector());
+        }
+        if (testTask.getPerformanceInterval() > 0) {
+            for (PerformanceTestSpec performanceTestSpec : testTask.getPerformanceTestSpecList()) {
+                performanceInspectionService.startInspectPerformanceTimer(performanceTestSpec, testTask.getPerformanceInterval());
+            }
+        }
+
         try {
-            run(deviceInfo, testTask, deviceTestTask);
+            run(deviceInfo, testTask, deviceTestTask, performanceInspectionService);
         } catch (Exception e) {
             deviceTestTask.getLogger().error(deviceInfo.getSerialNum() + ": " + e.getMessage(), e);
             saveErrorSummary(deviceTestTask, e);
         } finally {
+            List<PerformanceResult<?>> performanceResults = performanceInspectionService.parse();
             tearDown(deviceInfo, testTask, deviceTestTask);
         }
```

Comment:
Key codes of setup `PerformanceInspectionService`, starting performance inspection, and result analysis in `TestRunner`